.github/workflows/hatch_ci.yml

```diff
@@ -70,27 +70,43 @@ jobs:
         run: |
           hatch test --python ${{ inputs.max-python-version }}  --cover --parallel --junitxml=junit-${{ inputs.max-python-version }}.xml
 
-      - name: Report Test Results
+      - name: Status Check - Unit Tests on default python version
         uses: mikepenz/action-junit-report@v5
-        if: (hashFiles('junit-*.xml') != '') && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository)
+        if: hashFiles('junit-default.xml') != ''
         with:
-          check_name: Unit Test Results
-          report_paths: junit-*.xml
+          check_name: Unit Tests on default python version
+          report_paths: junit-default.xml
+
+      - name: Status Check - Unit Tests on Python ${{ inputs.min-python-version }}
+        uses: mikepenz/action-junit-report@v5
+        if: hashFiles('junit-${{ inputs.min-python-version }}.xml') != ''
+        with:
+          check_name: Unit Tests on Python ${{ inputs.min-python-version }}
+          report_paths: junit-${{ inputs.min-python-version }}.xml
+
+      - name: Status Check - Unit Tests on Python ${{ inputs.max-python-version }}
+        uses: mikepenz/action-junit-report@v5
+        if: hashFiles('junit-${{ inputs.max-python-version }}.xml') != ''
+        with:
+          check_name: Unit Tests on Python ${{ inputs.max-python-version }}
+          report_paths: junit-${{ inputs.max-python-version }}.xml
 
       - name: Combine Coverage Reports
         continue-on-error: true
+        if: hashFiles('*.cov') != ''
         run: |
           hatch run hatch-test.py${{ inputs.max-python-version }}:coverage combine
         shell: bash
 
       - name: Report coverage
         continue-on-error: true
+        if: hashFiles('*.cov') != ''
         run: |
           hatch run hatch-test.py${{ inputs.max-python-version }}:coverage xml
         shell: bash
 
       - name: SonarQube Scan
-        if: always() && github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository
+        if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository)
```

Comment:
We will run SonarQube when the PR branch is coming from the same organization.
For forked repository, we will need to explore on how to run scans for public contributions before they merge into SolaceLabs.